src/main/java/htsjdk/tribble/readers/TabixReader.java

```diff
@@ -439,7 +439,7 @@ public String next() throws IOException {
     public Iterator query(final int tid, final int beg, final int end) {
         TPair64[] off, chunks;
         long min_off;
-        if(tid< 0 || tid>=this.mIndex.length) return EOF_ITERATOR;
+        if (tid < 0 || beg < 0 || end < 0 || tid >= this.mIndex.length) return EOF_ITERATOR;
```

Comment:
I think this should be `end <= 0`, since the coordinates are 0-based closed-open.